src/main.rs

```diff
@@ -210,37 +209,60 @@ fn to_ca_certs(ca_certs_pem: &Vec<Vec<u8>>) -> Result<rustls::RootCertStore> {
     Ok(root_cert)
 }
 
-fn init_tracing(path: &Path) -> Result<WorkerGuard> {
-    if !path.exists() {
-        bail!("Path not found {path:?}");
-    }
-
+/// Initialize the tracing subscriber.
+///
+/// If `some_dir` is `None` or the runtime is in debug mode, logs will be printed to stdout.
+///
+/// Returns a vector of `WorkerGuard` that flushes the log when dropped.
+fn init_tracing(some_dir: Option<&Path>) -> Vec<WorkerGuard> {
+    let mut guards = vec![];
+    let subscriber = tracing_subscriber::Registry::default();
     let file_name = format!("{}.log", env!("CARGO_PKG_NAME"));
-    if std::fs::File::create(path.join(&file_name)).is_err() {
-        bail!("Cannot create file. {}/{file_name}", path.display());
-    }
 
-    let file_appender = tracing_appender::rolling::never(path, file_name);
-    let (file_writer, guard) = tracing_appender::non_blocking(file_appender);
-
-    let layer_file = fmt::Layer::default()
-        .with_ansi(false)
-        .with_target(false)
-        .with_writer(file_writer)
-        .with_filter(
-            EnvFilter::builder()
-                .with_default_directive(LevelFilter::INFO.into())
-                .from_env_lossy(),
-        );
+    let is_valid_file =
+        matches!(some_dir, Some(path) if std::fs::File::create(path.join(&file_name)).is_ok());
 
-    let layered_subscriber = tracing_subscriber::registry().with(layer_file);
-    #[cfg(debug_assertions)]
-    let layered_subscriber = layered_subscriber.with(
-        fmt::Layer::default()
-            .with_ansi(true)
-            .with_filter(EnvFilter::from_default_env()),
+    assert!(
+        some_dir.is_none() || is_valid_file,
+        "log_dir was specified but failed to create the log file in the directory: {}",
+        some_dir.expect("verified by negating is_none").display()
```

Comment:
I personally think it would be beneficial to handle the logging policy change in a separate PR. However, if you prefer to include it in this current PR, it might make sense to either create a separate issue specifically for item 2-2 of #117 or, at the very least, link #117 as related.

If you do decide to handle the logging policy change in this PR, please stay mindful of any last-minute changes that may arise from ongoing discussions around this logging policy.

Reply:
Since the logging policy hasn't been decided yet, it would be better to put it on hold. It's causing more confusion.
I'll delete the code and notify you again.

Reply:
The code has been removed.

Reply:
Got it, thank you!